File: 0443-string-compression/0443-string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        result, idx = 0, 0

        while idx < len(chars):
            target = chars[idx]
            cnt = 0
            while idx < len(chars) and chars[idx] == target:
                cnt += 1
                idx += 1
            chars[result] = target
            result += 1
            if cnt > 1:
                for c in str(cnt):
                    chars[result] = c
                    result += 1

        return result

File: 0443-string-compression/0443-string-compression.java
class Solution {
    public int compress(char[] chars) {
        int result = 0, idx = 0;

        while (idx < chars.length) {
            char target = chars[idx];
            int cnt = 0;

            while (idx < chars.length && chars[idx] == target) {
                idx++;
                cnt++;
            }

            chars[result] = target;
            result++;

            if (cnt > 1) {
                for (char c: String.valueOf(cnt).toCharArray()) {
                    chars[result] = c;
                    result++;
                }
            }
        }

        return result;
    }
}